### PR TITLE
Consider adding Artifact support

### PR DESCRIPTION
**Pull request by pitbulk**
Once we consider to add support for Artifact binding, take a look on this approach.

### CODE REVIEW

File: lib/Saml2/Auth.php [hunk 1]
```diff
@@ -113,13 +120,80 @@ public function processResponse($requestId = null)
...
+        $xml = <<<ARS
+<?xml version="1.0"?>
+<soap-env:Envelope xmlns:soap-env="http://schemas.xmlsoap.org/soap/envelope/"><soap-env:Header/>
+    <soap-env:Body>$rootXml</soap-env:Body>
+</soap-env:Envelope>
+ARS;
+
+        // POST the data using cURL
+        // TODO support basic auth
+        $ch = curl_init();
```
Comment (Hikariii) [inline lib/Saml2/Auth.php]: I consider calling curl directly a bad practice.
As php 5.3 support is required I would recommend using Guzzle 3 https://github.com/guzzle/guzzle3.
Why use curl when php has its own soap client http://php.net/manual/en/class.soapclient.php?

**Comment by madmatt** [comment]
Hey @pitbulk, is there any timeframe for artifact binding being added to php-saml? We're pretty keen to use this module, but the IdP we're integrating with only supports artifact binding and not POST binding :(

Thanks!

**Comment by pitbulk** [comment]
No plans on adding artifact binding in the near future.

**Comment by madmatt** [comment]
Thanks for your candour :)

File: lib/Saml2/Auth.php [hunk 2]
```diff
@@ -113,13 +120,80 @@ public function processResponse($requestId = null)
...
+
+        if ($out === false) {
+            throw new OneLogin_Saml2_Error(
+                'Could not retrieve ARS response.',
+                OneLogin_Saml2_Error::SAML_ARS_RESPONSE_INVALID
+            );
+        }
+
+        $response = new OneLogin_Saml2_ArtifactResponse($this->_settings, $out);
+        if (!$response->isSuccessful()) {
```
Comment (shoaibali) [inline lib/Saml2/Auth.php]: May I also suggest looking for Fault string in soap response ./soap-env:faultstring

File: lib/Saml2/ArtifactResolve.php [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+
+        $xmlSig->addReferenceList(array($this->_root), $type,
+            array('http://www.w3.org/2000/09/xmldsig#enveloped-signature', XMLSecurityDSig::EXC_C14N),
+            array('id_name' => 'ID', 'overwrite' => FALSE)
+        );
+
+        $xmlSig->sign($key);
+
+        if ($cert = $settings->getSPcert()) {
+            $xmlSig->add509Cert($cert, true);
```
Comment (shoaibali) [inline lib/Saml2/ArtifactResolve.php]: Is this doing mutual ssl ? i.e 3-way handshake which involves verifying the peer (client) as well.

**Pull request closed without merging**